src/prop_name.rs

```diff
@@ -0,0 +1,312 @@
+use crate::ffi_util::CStrLike;
+
+use std::ffi::{CStr, CString};
+
+/// A borrowed name of a RocksDB property.
+///
+/// The value is guaranteed to be a nul-terminated UTF-8 string.  This means it
+/// can be converted to [`CStr`] and [`str`] at zero cost.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
+pub struct PropName(CStr);
+
+impl PropName {
+    /// Creates a new object from a nul-terminated string with no internal nul
+    /// bytes.
+    ///
+    /// Panics if the `value` isn’t terminated by a nul byte or contains
+    /// interior nul bytes.
+    pub(crate) const fn new_unwrap(value: &str) -> &Self {
+        let bytes = if let Some((&0, bytes)) = value.as_bytes().split_last() {
+            bytes
+        } else {
+            panic!("input was not nul-terminated");
+        };
+
+        let mut idx = 0;
+        while idx < bytes.len() {
```

Comment:
Here should be `bytes.len() - 1`? We don't need to check the last element because it should be `\0`?

Reply:
`bytes` comes from `split_last` so it already has the terminating byte excluded.